ipamutils/utils.go

```diff
@@ -13,18 +15,34 @@ var (
 	// PredefinedGranularNetworks contains a list of 64K IPv4 private networks with host size 8
 	// (10.x.x.x/24) which do not overlap with the networks in `PredefinedBroadNetworks`
 	PredefinedGranularNetworks []*net.IPNet
-
-	initNetworksOnce sync.Once
+	initNetworksOnce           sync.Once
 )
 
-// InitNetworks initializes the pre-defined networks used by the built-in IP allocator
-func InitNetworks() {
+// PredefinedPools represent a set of address pools with prefix length Size.
+// Each pool in the set is derived from the Base pool. Base is to be passed
+// in CIDR format.Currently we support only local scope networks
+// Example: a Base "10.10.0.0/16 with Size 24 will define the set of 256
+// 10.10.[0-255].0/24 address pools
+type PredefinedPools struct {
+	Base string `json:"base"`
+	Size int    `json:"size"`
+}
+
+// InitNetworks initializes the local predefined address pools
+// with the default values.
+func InitNetworks(defaultAddressPool []*PredefinedPools) {
 	initNetworksOnce.Do(func() {
-		PredefinedBroadNetworks = initBroadPredefinedNetworks()
+		if defaultAddressPool != nil {
+			if InitAddressPools(defaultAddressPool) != nil {
+				logrus.Error("InitAddressPools failed to initialize default address pool")
```

Comment:
Glad the error check is in, but shouldn't it report the error that InitAddressPools itself returned along with the context string here?  e.g. something like
```go
if err := InitAddressPools(defaultAddressPool); err != nil {
    logrus.Errorf("InitAddressPools failed BLAH.... %v", err)
```

Reply:
As we already print the error message inside InitAddressPool, we would be duplicating same error message.

Reply:
@selansen I don't see the error being returned from `InitAddressPool` being printed in the log through `logrus.Errorf`. The specific `fmt.errorf()` statements in `InitAddressPool` does not print the error - they are creating the error object that the caller should dump/process. The `err` should be passed to `logrus.Errorf` as @ctelfer-docker suggested above, right?